packages/starlight/utils/navigation.ts

```diff
@@ -134,8 +141,15 @@ function makeLink(
 	badge?: Badge,
 	attrs?: LinkHTMLAttributes
 ): Link {
-	if (!isAbsolute(href)) href = pathWithBase(href);
-	const isCurrent = href === ensureTrailingSlash(currentPathname);
+	href = formatPath(href, {
+		format: project.build.format,
+	});
+	currentPathname =
+		project.build.format === 'file'
+			? stripTrailingSlash(currentPathname)
+			: ensureTrailingSlash(currentPathname);
```

Comment:
One question here: `currentPathname` is passed from `Astro.url.pathname`, does Astro provide this already formatted according to `build.format` and `trailingSlash` settings or would we need to call `formatPath(currentPathname)`?

Reply:
Great question! I believe the reason it doesn't autoformat is to maintain continuity.

When someone requests `/base/reference.html/`, it forwards the URL exactly as provided.

You're correct that we might need to invoke `formatPath`. This ternary operator won't function when a trailing slash is specified. I might have to adjust `formatPath` to avoid adding the base path twice. The current implementation assumes that base is not present.